noir-projects/noir-protocol-circuits/crates/parity-lib/src/root/root_parity_inputs.nr

```diff
@@ -39,18 +41,25 @@ impl RootParityInputs {
     }
 
     fn verify_child_proofs(self) {
-        for i in 0..NUM_BASE_PARITY_PER_ROOT_PARITY {
-            assert(
-                self.children[i].verification_key.hash == self.children[0].verification_key.hash,
+        for i in 1..NUM_BASE_PARITY_PER_ROOT_PARITY {
+            // No need to check verification_key.key directly, as it's verified in bb that it produces the given hash.
+            // We ensure that all the keys are the same by asserting that their hashes are equal across all children.
```

Comment:
Thank you for adding a comment! I was thinking that otherwise it would be difficult to know what is actually constrained here or in the background